include/zephyr/pm/state.h

```diff
@@ -358,6 +364,34 @@ struct pm_state_constraint {
 			Z_PM_STATE_FROM_DT_CPU, (), node_id)		       \
 	}
 
+/**
+ * @brief initialize a device pm constraint with information from devicetree.
+ *
+ * @param node_id Node identifier.
+ */
+#define PM_STATE_CONSTRAINT_INIT(node_id)                                     \
+	{                                                                     \
+		.state = PM_STATE_DT_INIT(node_id),                           \
+		.substate_id = DT_PROP_OR(node_id, substate_id, 0),           \
+	}
+
+#define PM_STATE_CONSTRAINT_REF(node_id, phandle, idx)                        \
+	PM_STATE_CONSTRAINT_INIT(DT_PHANDLE_BY_IDX(node_id, phandle, idx))
+
+#define PM_STATE_CONSTRAINTS_LIST_NAME(node_id, phandles)                     \
+	node_id ## _ ## phandles ## _constraints
+
+#define PM_STATE_CONSTRAINTS_LIST_DEFINE(node_id, phandles)                                \
```

Comment:
replace `phandles` with `prop`, the user does not care what type the prop is when using a macro like this, just the name of the prop. Also add a bit of documentation to it